### PR TITLE
tfix and optimization for getProxyMode

### PR DESCRIPTION
**Pull request by Mashimiao**
Signed-off-by: Ma Shimiao <mashimiao.fnst@cn.fujitsu.com>

**What this PR does / why we need it**:
unify proxy to proxier, and warning for unknown proxymode in server_windows.go and several logic changes.

**Release note**:
```
None
```

### CODE REVIEW

File: cmd/kube-proxy/app/server_others.go [hunk 1]
```diff
@@ -247,20 +247,19 @@ func getProxyMode(proxyMode string, iptver iptables.IPTablesVersioner, kcompat i
...
 	}
-
-	if len(proxyMode) > 0 && proxyMode == proxyModeIPTables {
+	if proxyMode == proxyModeIPTables {
 		return tryIPTablesProxy(iptver, kcompat)
 	}
-
-	if utilfeature.DefaultFeatureGate.Enabled(features.SupportIPVSProxyMode) {
-		if proxyMode == proxyModeIPVS {
+	if proxyMode == proxyModeIPVS {
```
Comment (cmluciano) [inline cmd/kube-proxy/app/server_others.go]: Any chance you can refactor these options with a switch statement?

Comment (Mashimiao) [reply cmd/kube-proxy/app/server_others.go]: yes, updated. @cmluciano  PTAL

**Comment by Mashimiao** [comment]
ping @lavalamp  @dchen1107

**Comment by BenTheElder** [comment]
cross is pending a fix again unfortunately :-(

**Comment by BenTheElder** [comment]
/test pull-kubernetes-unit
this however is a flake: https://github.com/kubernetes/kubernetes/issues/56262

**Pull request closed without merging**